src/App.tsx

```diff
@@ -1,39 +1,89 @@
-import React from 'react';
+import { Component } from 'react';
 import './App.scss';
+import {
+  Button,
+  Paper,
+  Box,
+} from '@mui/material';
+import { Clock } from './Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+interface State {
+  hasClock: boolean;
+  clockName: string;
+}
+
+export class App extends Component<{}, State> {
+  state: State = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  timerId = 0;
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+  componentDidMount() {
+    this.timerId = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+  }
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  componentDidUpdate(_: {}, prevState: State) {
+    const { clockName, hasClock } = this.state;
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+    if (clockName !== prevState.clockName && hasClock) {
+      window.console.debug(`Renamed from ${prevState.clockName} to ${clockName}`);
+    }
+  }
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+  componentWillUnmount() {
+    window.clearInterval(this.timerId);
+  }
 
-        {' time is '}
+  render() {
+    const { clockName, hasClock } = this.state;
 
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
+    return (
+      <div className="app">
```

Comment:
button is jumping

<img width="442" alt="image" src="https://user-images.githubusercontent.com/91411692/199333238-d452c114-62c3-4f1d-a48d-71c83c4caefc.png">
<img width="440" alt="image" src="https://user-images.githubusercontent.com/91411692/199333279-989413cf-82a0-4b6a-be8b-eb5707ebe726.png">